src/event-handler/on.ts

```diff
@@ -19,10 +19,16 @@ function handleEventHandlers(
 }
 export function receiverOn(
   state: State,
-  webhookNameOrNames: EmitterWebhookEventName | EmitterWebhookEventName[],
+  webhookNameOrNames:
+    | EmitterWebhookEventName
+    | readonly EmitterWebhookEventName[],
   handler: Function
 ) {
-  if (Array.isArray(webhookNameOrNames)) {
+  if (
+    (Array.isArray as (arg: any) => arg is readonly unknown[])(
```

Comment:
Not a fan of this - I don't think `readonly` gives us much advantages to be worth needing to cast like this, nor to have extra runtime code.

Reply:
What's the best workaround?

Reply:
@G-Rath

Reply:
@jablko is there a specific use case where you find these changes valuable? Maybe with the example you would like to get type-covered we can come up together with other alternatives.

Reply:
> What's the best workaround?

There isn't really a workaround.